cincinnati/src/plugins/internal/edge_add_remove.rs

```diff
@@ -423,4 +434,81 @@ mod tests {
 
         assert_eq!(expected_graph, processed_graph);
     }
+
+    /// Ensure the label processing order is consistent with the documentation
+    #[test]
+    fn ensure_label_processing_order() -> Fallible<()> {
+        let key_prefix = "test_prefix".to_string();
+
+        let metadata: HashMap<usize, HashMap<String, String>> = [
+            (
+                0,
+                [(
+                    format!("{}.{}", key_prefix, "next.add"),
+                    String::from("1.0.0"),
+                )]
+                .iter()
+                .cloned()
+                .collect(),
+            ),
+            (
+                1,
+                [
+                    (
+                        format!("{}.{}", key_prefix, "previous.remove"),
+                        String::from("0.0.0"),
+                    ),
+                    (
+                        format!("{}.{}", key_prefix, "next.add"),
+                        String::from("2.0.0"),
```

Comment:
I may be misunderstanding something. Shouldn't this be supposed to cover (i.e. re-add) the same edge that the directive above removes?

Reply:
This label is contradicted by the subsequent label at https://github.com/openshift/cincinnati/pull/127/files#diff-0b0028d0d2d6c8d8c41ae291b2dce0ebR474.

I could add more contradictions to an extent where every label has one. Would you prefer that?

Reply:
I see.
My understanding was that this was only covering contradicting metadata processing **for each node**, while the case seems to relies on contradicting metadata across **two different nodes**. Shallow reading from my side, sorry.

This test is fine then, maybe an hint in the docstring would make the above obvious.

Reply:
That is an interesting twist which I didn't intend to cover in this test. I will go back and think about incorporating that one too. I'll add a note to the docstring either way.